core/client/app/routes/posts.js

```diff
@@ -69,6 +69,22 @@ export default AuthenticatedRoute.extend(ShortcutsRoute, PaginationRouteMixin, {
     },
 
     actions: {
+        willTransition(transition) {
+            let transitionDueToDeletion = this.get('controller.currentPost.isDeleted')
+                && transition.targetName === 'posts.index';
+            let posts, nextPost, deletedPostIndex;
+
+            if (transitionDueToDeletion) {
+                posts = this.get('controller.sortedPosts');
+                deletedPostIndex = this.get('controller.currentPostIndex');
+                nextPost = posts.objectAt(deletedPostIndex);
+                if (nextPost) {
+                    return this.transitionTo('posts.post', nextPost);
```

Comment:
I actually had that originally, and took it out on the hunch that it might be the issue :D Read up on their docs: the difference is, if you call `.abort()` within here you'll trigger a new round of "willTransition", whereas just doing a `transitionTo` is equivalent to a redirect so you won't call willTransition twice